### PR TITLE
Skip forbidden resources as they are rarely required

### LINKED ISSUE #14: Allow skipping resources which are not accessible with current permission
@ahmetb mentioned in one of his comment about introducing `--ignore-inaccessible`
https://github.com/ahmetb/kubectl-tree/issues/9#issuecomment-570949171

I find this very useful feature as most of the users don't have access to prod env resources and the tree command just explodes.

### LINKED ISSUE #60: Fails if privileges are limited
In a production environment, users usually have limited privileges. In this case `kubectl tree <kind> <name>` fails with
```
Error: error while querying api objects: listing resources failed (<GVK>): <kind> is forbidden
```

I  would suggest to print a warning for 403s but continue processing otherwise.

EDIT: Just saw there is already https://github.com/ahmetb/kubectl-tree/pull/55 for this. Please merge.

### PR DESCRIPTION
**Pull request by alam0rt**
In a multi-tenant cluster, rarely do users have permissions to view all API resources. This makes it so any resource that is forbidden to the user will be skipped.

Fixes #14 
Fixes #60

### CODE REVIEW

**Comment by ahmetb** [comment]
I think this is agreeable, there's just a little bug that we need to assign to `errResult` only if it's not a Forbidden error. I'll push that commit and merge.

**Comment by yutkin** [comment]
Any updates on that?

**Comment by rkulow** [comment]
+1, would be nice to see this merged.

**Comment by Novemb3r** [comment]
+1 waiting for it

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -11,7 +11,7 @@ jobs:
     - name: Setup Go
       uses: actions/setup-go@v1
       with:
-        go-version: 1.18
+        go-version: 1.20
```
Comment (ahmetb) [inline .github/workflows/ci.yml:14]: You need to type ' or " around this otherwise it becomes 1.2 float :) at this point let's do 1.22

**Review by ahmetb — APPROVED** [review]
Sorry for the delay. Only one small comment.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (ahmetb) [reply .github/workflows/ci.yml:14]: Omg i made this error myself it turns out 🤦

**Comment by initor** [comment]
+1, would be nice to see this merged.

**Comment by alam0rt** [comment]
sorry for the delay - missed this.

hows this look?

**Comment by dacofr** [comment]
+1
Hello, When do you think this fix can be merged ?
Thanks

**Review by ahmetb — APPROVED** [review]

**Comment by ahmetb** [comment]
it seems `--rm-dist` flag got removed from goreleaser, in favor of `--clean`.

**Pull request merged**